Program.cs

```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
                         AddCharacter();
                         break;
                     case 2:
-                        AddParty();
+                        ShowCharacters();
                         break;
                     case 3:
-                        AddEncounter();
+                        AddParty();
                         break;
                     case 4:
-                        StartEncounter();
+                        AddEncounter();
                         break;
                     case 5:
+                        StartEncounter();
+                        break;
+                    case 6:
                         Quit();
                         break;
                     default:
                         break;
                 }
             }
-          
         }
 
-
         static void AddCharacter()
         {
-            Menu CharacterMenu = new Menu("Choose an Option", new string[] { "Enter Name", "Enter HP", "Enter Damage" });
+            bool goBack = false;
+            Menu CharacterMenu = new Menu("Add a character", new string[] {
+                "Enter Name",
+                "Enter HP",
+                "Enter Damage",
+                "Show Character",
+                "Go Back"
+            });
 
-            while (!goBack) 
-            {
+            Character newCharacter = new Character();
+            Characters.Add(newCharacter);
 
+            while (!goBack)
+            {
                 CharacterMenu.Print();
                 int option = CharacterMenu.ReadOption();
 
                 switch (option)
                 {
                     case 1:
-                        EnterName();
+                        EnterName(newCharacter);
```

Comment:
We now pass the newly created character to the functions that will update the various properties.